flow360/component/simulation/validation/validation_context.py

```diff
@@ -185,7 +185,7 @@ def decorator(func: Callable):
         def wrapper(self: Any, *args, **kwargs):
             current_levels = get_validation_levels()
             # Run the validator only if the current levels matches the specified context or is ALL
-            if current_levels is None or any(lvl in (context, ALL) for lvl in current_levels):
+            if current_levels is not None and any(lvl in (context, ALL) for lvl in current_levels):
```

Comment:
that was not mistake. The motivation was the following:
if user is constructing SimulationParams and uses any parameter, we always validate this parameter. It is most likely optional or belongs to optional parent. User is free to remove it to pass validation. It is context irrelevant that any explicit inputs must be valid.

Thus we always run validator when running SimulationParams without validation context. But for developer user, for example web GUI integration or some python client functionalities, we can ignore some errors as we understand why some errors can be ignored. That is the motivation of this logic here.

Reply:
But then what is the point of `context_validator` other than providing the context of error? Or is this exactly the purpose?

Reply:
that is the purpose